_posts/2017-11-23-Rust-1.22.md

```diff
@@ -0,0 +1,166 @@
+
+## What's in 1.22.0 stable
+
+The headline feature for this release is one many have been waiting for
+for a long time: you can now [use `?` with `Option<T>`]! About a year
+ago, in [Rust 1.13], we introduced the `?` operator for working with
+`Result<T, E>`. Ever since then, there's been discussion about how far
+`?` should go: should it stay only bound to results? Should it be
+user-extensible? Should it be usable with `Option<T>`?
+
+In Rust 1.22, basic usage of `?` with `Option<T>` is now stable.
+This code will now compile:
+
+```rust
+fn try_option_some() -> Option<u8> {
+    let val = Some(1)?;
+    Some(val)
+}
+assert_eq!(try_option_some(), Some(1));
+
+fn try_option_none() -> Option<u8> {
+    let val = None?;
+    Some(val)
+}
+assert_eq!(try_option_none(), None);
+```
+
+However, this functionality is still a bit limited; you cannot yet write
+code that mixes results and options with `?` in the same function, for
+example. In the future, this will be possible, and already is in nightly
+Rust; expect to hear more about this in a future release.
+
+[use `?` with `Option<T>`]: https://github.com/rust-lang/rust/pull/42526
+[Rust 1.13]: https://blog.rust-lang.org/2016/11/10/Rust-1.13.html
+
+Types that implement `Drop` are [now allowed in `const` and `static`
+items](https://github.com/rust-lang/rust/pull/44456). Like this:
+
+```rust
+struct Foo {
+    a: u32
+}
+
+impl Drop for Foo {
+    fn drop(&mut self) {}
+}
+
+const F : Foo = Foo { a : 0 };
+static S : Foo = Foo { a : 0 };
+```
+
+On its own, this change doesn't bring much, but as we improve our
+ability to compute things at compile-time, more and more will be
+possible in `const` and `static`.
+
+Additionally, some small quality-of-life improvements:
+
+[Two] recent [compiler changes] should speed up compiles in debug mode. We
+don't have any specific numbers to commit to with these changes, but as
+always, compile times are very important to us, and we're continuing to
+work on improving them.
+
+[Two]: https://github.com/rust-lang/rust/pull/45075
+[compiler changes]: https://github.com/rust-lang/rust/pull/45064
+
+[`T op= &T` now works for primitive types][add], which is a fancy way of saying:
+
+```rust
+let mut x = 2;
+let y = &8;
+
+// this didn't work, but now does
+x += y;
+```
+
+Previously, you'd have needed to write `x += *y` in order to de-refernece, so
+this solves a small papercut.
+
+[add]: https://github.com/rust-lang/rust/pull/44287
+
+[Backtraces are improved on MacOS](https://github.com/rust-lang/rust/pull/44251).
+
+You can now [create `compile-fail` tests in Rustdoc], like this:
+
+```
+/// ```compile_fail
+/// let x = 5;
+/// x += 2; // shouldn't compile!
+/// ```
+```
+
+Please note that these kinds of tests can be more fragile than others, as
+sometimes, additions to Rust cause code that previously wouldn't have compiled
+into code that does. Consider the first announcement with `?`, for example:
```

Comment:
This sentence is a little tough. How about "Please note that these kinds of tests can be more fragile than others, as additions to Rust may cause code to compile when it previously would not"?